Dockerfile

```diff
@@ -1,31 +1,64 @@
 # Dockerfile for zhmc-prometheus-exporter project
 #
 # This image runs the zhmc_prometheus_exporter command.
-# The standard metric definition file is provided in its default location, so
-# the -m option does not need to be specified.
 #
-# The HMC credentials file still needs to be made available to the container
+# The exporter config file needs to be made available to the container
 # using some mount option and specified with -c.
 #
 # Example docker command to run the exporter using a locally built version of this image:
 #
-#   docker run -it --rm -v $(pwd)/myconfig:/root/myconfig -p 9291:9291 zhmcexporter -c /root/myconfig/config.yaml -v
+#   docker run -it --rm -v $(pwd)/myconfig:/root/myconfig -p 9291:9291 zhmcexporter:2.0.0 -c /root/myconfig/config.yaml -v
 
-FROM python:3.9-slim
+FROM python:3.12-slim as builder
 
-# Install this package
-ENV TMP_DIR=/tmp/zhmc-prometheus-exporter
-WORKDIR $TMP_DIR
-COPY . $TMP_DIR
+# Path name of binary distribution archive of zhmc-prometheus-exporter package
+ARG bdist_file
+RUN : "${bdist_file:?Build argument bdist_file is required}"
 
-# TODO: Remove git install again once PR https://github.com/prometheus/client_python/pull/946 is released on Pypi
-RUN apt-get update
-RUN apt-get install -y --no-install-recommends git
+# Install git in case the requirements use git+https links.
+RUN apt-get update \
+  && apt-get install -y --no-install-recommends git
 
-RUN pip install . && rm -rf $TMP_DIR
+# Install the zhmc-prometheus-exporter package
+COPY ${bdist_file} /tmp/${bdist_file}
+RUN pip install --user /tmp/${bdist_file} \
+  && rm -f /tmp/${bdist_file} \
+  && pip list
 
-# Set the current directory when running this image
-WORKDIR /root
+# Uninstall Python packages not needed for running the package
+RUN python -m pip uninstall -y pip setuptools wheel
+
+FROM python:3.12-alpine
+
+# Version of the zhmc-prometheus-exporter package
+ARG package_version
+RUN : "${package_version:?Build argument package_version is required}"
+
+# Image build date in ISO-8601 format
+ARG build_date
+RUN : "${build_date:?Build argument build_date is required}"
+
+# Git commit ID of the zhmc-prometheus-exporter repo used to build the image
+ARG git_commit
+RUN : "${git_commit:?Build argument git_commit is required}"
+
+# Set image metadata
+LABEL org.opencontainers.image.title="IBM Z HMC Prometheus Exporter"
+LABEL org.opencontainers.image.version="${package_version}"
+LABEL org.opencontainers.image.authors="Andreas Maier, Kathir Velusamy"
+MAINTAINER Andreas Maier, Kathir Velusamy
```

Comment:
I'm just a newbie to this docker, from whatever I read from web, LABEL org.opencontainers.image.authors is the replacement for the MAINTAINER( this is deprecated in one of the releases), so my question is do we need both of them in our Dockerfile.

Reply:
Agreed. I have removed MAINTAINER.